eZ/Publish/Core/Repository/Permission/PermissionsCriterionResolver.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\Repository\Permission;
+
+use eZ\Publish\API\Repository\PermissionCriterionResolver as APIPermissionCriterionResolver;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion\LogicalAnd;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion\LogicalOr;
+use eZ\Publish\API\Repository\Values\User\Limitation;
+use eZ\Publish\API\Repository\PermissionResolver as PermissionResolverInterface;
+use eZ\Publish\Core\Repository\Helper\LimitationService;
+use RuntimeException;
+
+/**
+ * Implementation of Permissions Criterion Resolver.
+ *
+ * @since 6.12
+ */
+class PermissionsCriterionResolver implements APIPermissionCriterionResolver
+{
+    /**
+     * @var \eZ\Publish\API\Repository\PermissionResolver
+     */
+    private $permissionResolver;
+
+    /**
+     * @var \eZ\Publish\Core\Repository\Helper\LimitationService
+     */
+    private $limitationService;
+
+    /**
+     * Constructor.
+     *
+     * @param \eZ\Publish\API\Repository\PermissionResolver $permissionResolver
+     * @param \eZ\Publish\Core\Repository\Helper\LimitationService $limitationService
+     */
+    public function __construct(
+        PermissionResolverInterface $permissionResolver,
+        LimitationService $limitationService
+    ) {
+        $this->permissionResolver = $permissionResolver;
+        $this->limitationService = $limitationService;
+    }
+
+    /**
+     * Get content-read Permission criteria if needed and return false if no access at all.
+     *
+     * @uses \eZ\Publish\API\Repository\PermissionResolver::hasAccess()
```

Comment:
For consistency's sake we should also hint `@uses` for `getCurrentUserReference`